src/accelerate/big_modeling.py

```diff
@@ -324,63 +324,67 @@ def dispatch_model(
             called directly during the forward, for instance if a `dense` linear layer is registered, but at forward,
             `dense.weight` and `dense.bias` are used in some operations instead of calling `dense` directly.
     """
-    if not is_torch_version(">=", "1.9.0"):
-        raise NotImplementedError("Model dispatching requires torch >= 1.9.0")
     # Error early if the device map is incomplete.
     check_device_map(model, device_map)
 
-    if main_device is None:
-        if set(device_map.values()) == {"cpu"} or set(device_map.values()) == {"cpu", "disk"}:
-            main_device = "cpu"
+    # We attach hooks if the device_map have at least 2 different devices. Otherwise, the model in already loaded
+    # in the unique device and the user can decide where to dispatch the model.
+    if len(device_map.values()) > 1:
```

Comment:
I think it's missing an else with `model.to()` the only device in the dict.

Reply:
Oh yeah, we can add that to be sure. However, the model should already be on the right device if the user uses `load_checkpoint_in_model` or load the model from` from_pretrained` method

Reply:
Yes, but the user might be calling `dispatch_model` by themselves.